src/playground-code-editor.ts

```diff
@@ -182,17 +182,16 @@ export class PlaygroundCodeEditor extends LitElement {
     this.requestUpdate('value', oldValue);
   }
 
+  @property()
+  documentKey?: string;
+
   /**
-   * Alternative to `value`, and takes precedence over `value`. Use `docCache`
-   * to pass a CodeMirror document instance directly. If `docCache` is set,
-   * `value` will no longer update the contents of the editor. The document
-   * instance should be used if history needs to be preserved.
+   * Will keep 50 document instances before exhibiting LRU cache behavior. On
+   * reaching max size, the least recently used document history will be lost.
    *
-   * Whenever a new document instance is created, a `newdocinstance` event is
-   * fired containing a `codeMirrorDoc` property.
+   * This prevents memory leaks and unbounded growth of the cache.
    */
-  @property({attribute: false})
-  docCache?: Doc;
+  private readonly _privateDocCache = new Map<string, Doc>();
```

Comment:
Sorry not sure if this was actually intended to be reviewed or not, but why not use a WeakMap here?

Reply:
I could not seem to get the file objects to cache hit against a WeakRef. Going to try the WeakRef again because now I have this working and a manual test via configurator works well, and other places have been cleaned up.

Reply:
SMH, it works. I must have had some of the previous implementation causing an issue.